types/apiResponse.ts

```diff
@@ -0,0 +1,18 @@
+export enum StatusMessageType {
+  Error = 0,
+  Warning = 1,
+  Information = 2,
+  Success = 3,
+}
+
+export interface StatusMessage {
+  type: StatusMessageType;
+  message: string;
+}
+
+export interface ApiResponse<T> {
+  payload: T | EmptyPayload;
+  messages: StatusMessage[];
+}
+
+export type EmptyPayload = Record<string, never>;
```

Comment:
Ooh what's this and how is it different from `{}`? (Just curious)

Reply:
I used this at first then the linters forced me to change to this lol